Sell/Basket/Repository/BasketError.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  BasketError.swift
+//  RealifeTech
+//
+//  Created by Mickey Lee on 24/06/2021.
+//  Copyright © 2021 Realife Tech. All rights reserved.
+//
+
+import Foundation
+
+public enum BasketError: Error, LocalizedError {
+
+    case outOfStock
+    case priceChange
+    case mixedBasket
+    case emptyBasket
+    case regularErrors([Error])
```

Comment:
Possibly remove this case (see below comment on `func transformErrorIfNeccessary(_) -> BasketError? `